### ISSUE TITLE
FR: Generate snapshots based on Zones

### ISSUE THREAD

**Issue by scstraus** [blakeblackshear/frigate#269]
Hi Blake,

It would be great if we could subscribe to the MQTT topic 

```frigate/<zone_name>/<object_name>/snapshot```

To get snapshots of zones. My zones are working well, now I want notifications showing who was in them :-).

**Comment by scstraus** [comment]
I’m still here stalebot!

**Comment by paularmstrong** [comment]
What are you using to process your events? I subscribe in Node-RED and have a function filter to parse out the message from `frigate/<camera_name>/<object_name>/snapshot` and take action on various conditions that way.

**Comment by scstraus** [comment]
Yeah actually this is a v.8 feature so I will close it.

**Comment by scstraus** [comment]
Actually I was wrong, you can get a sensor for the zone, but not a snapshot feed for it (at least the cameras set up by the hass component don't give theat). As I would like to show which persons were last in a specific zone for a camera but not people from other zones on that camera, a sensor is not enough (at least not without quite a bit of extra work on my side).

**Comment by scstraus** [comment]
Or another even better  way to solve the problem of displaying the right thing in the Lovelace UI based on the activated zone would be if there was some way to reference the exact snapshot that went with a certain event. This would also solve the problem of getting a different snapshot in your notification if you open it after too long and another event happened on that camera since the notification was sent. I can’t think of how that would be done though.

**Comment by blakeblackshear** [comment]
That's what the events mqtt topic is for. See the notification section of the docs for how to get and use the event_id for this.

**Comment by scstraus** [comment]
Thanks Blake. It seems as though I haven't fully digested all the new features of 0.8 yet. They are very extensive.

I have added this to one of my zone notification automations and I will see how it works. It looks very promising if it can give the exact matching snapshot of the event which I got the notification for without changing later on. I have to admit I don't entirely understand how it works yet, I will have to watch the events and see what they look like.

One followup question. Can such a mechanism also be turned into a camera feed for a zone so that I can display the most recent person in a specific zone as a camera feed in lovelace too?

**Comment by scstraus** [comment]
I tried that automation, but on iOS I get:
```
No entity_id found in payload!
```
It seems that iOS doesn't support the "image" parameter and so an entity_id needs to be provided for the image. Am I out of luck on iOS or is there a way to get an entity_id that can serve up that image?

**Comment by scstraus** [comment]
So I tried it out, but basically I got the same results I got with the camera feed which just gives me the latest snapshot regardless of how old the notification was. It does not appear to correctly be able to refer to older events that correspond to the event that creates the notification. I have opened a [new issue here](https://github.com/blakeblackshear/frigate/issues/871) rather than clutter up this one more.
